src/gui/src/layoutViewer.cpp

```diff
@@ -474,6 +611,192 @@ void LayoutViewer::updateRubberBandRegion()
   update(rect.right() - unit / 2, rect.top(), unit, rect.height());
 }
 
+LayoutViewer::Edges LayoutViewer::searchNearestEdge(const std::vector<Search::Box>& boxes, const odb::Point& pt, bool& ok)
+{
+  // find closest edges for each object returned
+  std::vector<Edges> candidates;
+  for (auto& box : boxes) {
+    auto pt0 = box.min_corner();
+    auto pt1 = box.max_corner();
+
+    Edge vertical;
+    // vertical edges
+    if (std::abs(pt.x() - pt0.get<0>()) < std::abs(pt.x() - pt1.get<0>())) {
+      // closer to pt0
+      vertical.first = odb::Point(pt0.get<0>(), pt0.get<1>());
+      vertical.second = odb::Point(pt0.get<0>(), pt1.get<1>());
+    } else {
+      // closer to pt1
+      vertical.first = odb::Point(pt1.get<0>(), pt0.get<1>());
+      vertical.second = odb::Point(pt1.get<0>(), pt1.get<1>());
+    }
+
+    Edge horizontal;
+    // horizontal edges
+    if (std::abs(pt.y() - pt0.get<1>()) < std::abs(pt.y() - pt1.get<1>())) {
+      // closer to pt0
+      horizontal.first = odb::Point(pt0.get<0>(), pt0.get<1>());
+      horizontal.second = odb::Point(pt1.get<0>(), pt0.get<1>());
+    } else {
+      // closer to pt1
+      horizontal.first = odb::Point(pt0.get<0>(), pt1.get<1>());
+      horizontal.second = odb::Point(pt1.get<0>(), pt1.get<1>());
+    }
+
+    candidates.push_back({horizontal, vertical});
+  }
+
+  if (candidates.empty()) {
+    ok = false;
+    return Edges();
+  }
+
+  // find closest edge overall, start with last one
+  Edges closest_edges = candidates.back();
+  candidates.pop_back();
+  for (auto& edge_set : candidates) {
+    // vertical edges
+    if (std::abs(pt.x() - edge_set.vertical.first.x()) < std::abs(pt.x() - closest_edges.vertical.first.x())) {
+      closest_edges.vertical = edge_set.vertical;
+    }
+
+    // horizontal edges
+    if (std::abs(pt.y() - edge_set.horizontal.first.y()) < std::abs(pt.y() - closest_edges.horizontal.first.y())) {
+      closest_edges.horizontal = edge_set.horizontal;
+    }
+  }
+
+  ok = true;
+  return closest_edges;
+}
+
+LayoutViewer::Edge LayoutViewer::findEdge(const odb::Point& pt, bool horizontal, bool& ok)
```

Comment:
You can return a tuple of bool, edge rather than take ok as an arg.  On the calling side you can use

auto [ok, edge] = findEdge(...);